File: 백준/Silver/19941. 햄버거 분배/햄버거 분배.java
import java.util.Scanner;

class Main {
    private static int solution(int k, char[] arr) {

        int cnt = 0;
        int len = arr.length;

        for (int i = 0; i < len; i++) {
            if (arr[i] == 'P') {
                for (int j = i - k; j <= i + k; j++) {
                    if (j >= 0 && j < len && arr[j] == 'H') {
                        arr[j] = ' ';
                        cnt++;
                        break;
                    }
                }
            }
        }

        return cnt;
    }

    public static void main(String[] args) {
        Scanner kb = new Scanner(System.in);

        int n = kb.nextInt();
        int k = kb.nextInt();

        char[] arr = kb.next().toCharArray();

        System.out.println(solution(k, arr));

    }
}

File: 백준/Silver/19941. 햄버거 분배/햄버거 분배.py
import sys

def solution(k, arr):
    cnt = 0

    for i in range(len(arr)):
        if arr[i] == 'P':
            for j in range(i - k, i + k + 1):
                if 0 <= j and j < len(arr) and arr[j] == 'H':
                    cnt += 1
                    arr[j] = ' '
                    break

    return cnt


if __name__ == "__main__":
    n, k = map(int, input().split())
    arr = list(sys.stdin.read())

    print(solution(k, arr))
